app/controllers/courses_controller.rb

```diff
@@ -197,6 +197,9 @@ def notify_untrained
   def delete_all_weeks
     @course = find_course_by_slug(params[:id])
     @course.weeks.destroy_all
+    if @course.approved? 
```

Comment:
These can be removed in the controllers, since there's now a check for this in the service object.